{{cookiecutter.project_slug}}/frontend/src/store/modules/main.ts

```diff
@@ -0,0 +1,252 @@
+import { api } from "@/api";
+import { getLocalToken, removeLocalToken, saveLocalToken } from "@/utils";
+import router from "@/router";
+import { AxiosError } from "axios";
+import { VuexModule, Module, Mutation, Action } from "vuex-module-decorators";
+import { IUserProfile, IUserProfileUpdate, IAppNotification } from "@/interfaces";
+import { UNAUTHORIZED } from "http-status-codes";
+
+@Module({ name: "main" })
+export default class MainModule extends VuexModule {
+  token = "";
+  isLoggedIn: boolean | null = null;
+  logInError = false;
+  userProfile: IUserProfile | null = null;
+  dashboardMiniDrawer = false;
+  dashboardShowDrawer = true;
+  notifications: IAppNotification[] = [];
+
+  get hasAdminAccess() {
+    return (
+      this.userProfile && this.userProfile.is_superuser && this.userProfile.is_active
```

Comment:
Adding `this.isLoggedIn && ` to this function will fix a Bug where you log in as admin and then logout. The drawer will now still hold the `Admin` items (I disabled the mandatory login). This but could also be fixed in `removeLogIn()`, see below.

These changes wouldn't hurt any developer keeping the login guard but would definitely help those removing it.

Reply:
I would like to incorporate this change. Please make a change suggestion so I can accept it. While you're at it, you can add a little syntactic sugar by using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

Reply:
```suggestion
      this.isLoggedIn && this.userProfile?.is_superuser?.is_active
```
I'm not quite sure, if the third access to `is_active` works (can't test right now)

Reply:
> this.isLoggedIn && this.userProfile?.is_superuser?.is_active

Yes, this shouldn't work as it would mean `is_superuser` is an object or `undefined`, containing a `is_active` key, which is not the case.